test-suite/Math/NumberTheory/DiophantineTests.hs

```diff
@@ -33,8 +34,17 @@ cornacchiaBruteForce (Positive d) (Positive a) = gcd d m /= 1 || findSolutions [
             where x2 = m - d*y*y
                   x = integerSquareRoot x2
 
+linearTest :: (a ~ Integer) => a -> a -> a -> a -> Bool
+linearTest a b c k =
+  case solveLinear Lin {..} of
+    Nothing -> True -- Disproving this would require a counter example
```

Comment:
The problem is that `solveLinear = const Nothing` passes this test. Can we come up with a test which requires `solveLinear` return `Just{}` at least in some cases?